frontend/public/components/utils/k8s-watch-hook.ts

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector, useDispatch } from 'react-redux';
+import { Map as ImmutableMap } from 'immutable';
+import { createSelectorCreator, defaultMemoize } from 'reselect';
+import { makeQuery, makeReduxID } from './k8s-watcher';
+import * as k8sActions from '../../actions/k8s';
+import { K8sResourceKind, K8sKind } from '../../module/k8s';
+import { FirehoseResource } from './types';
+import { RootState } from '../../redux';
+
+export class NoModelError extends Error {
+  constructor() {
+    super('Model does not exist');
+  }
+}
+
+export const useK8sWatch = <R extends K8sResourceKind = K8sResourceKind>(
+  resources: FirehoseResource[] = [],
+): WatchK8sResult<R> => {
+  const k8sModelSelectorCreator = React.useMemo(
+    () =>
+      createSelectorCreator(
+        //specifying createSelectorCreator<ImmutableMap<string, any>> throws type error
+        defaultMemoize as <F extends Function>(
+          func: F,
+          equalityCheck?: (
+            a: ImmutableMap<string, any>,
+            b: ImmutableMap<string, any>,
+            index: number,
+          ) => boolean,
+        ) => F,
+        (oldModels, newModels) =>
+          resources.every(({ kind }) => oldModels.get(kind) === newModels.get(kind)),
+      ),
+    [resources],
+  );
+
+  const k8sModelSelector = React.useMemo(
```

Comment:
Same question here, `k8sModelSelector` is a function, so maybe `useCallback` ?

Reply:
Same as above, useMemo is correct here too. We want to memoize the result of calling `k8sModelSelectorCreator`